irr/src/CIrrDeviceSDL.cpp

```diff
@@ -1157,14 +1151,37 @@ bool CIrrDeviceSDL::isWindowMaximized() const
 
 bool CIrrDeviceSDL::isFullscreen() const
 {
+	if (!Window)
+		return false;
+	u32 flags = SDL_GetWindowFlags(Window);
+	return (flags & SDL_WINDOW_FULLSCREEN) != 0 ||
+			(flags & SDL_WINDOW_FULLSCREEN_DESKTOP) != 0;
+}
+
+u32 CIrrDeviceSDL::getFullscreenFlag(bool fullscreen)
+{
+	if (!fullscreen)
+		return 0;
 #ifdef _IRR_EMSCRIPTEN_PLATFORM_
-	return SDL_GetWindowFlags(0) == SDL_WINDOW_FULLSCREEN;
+	return SDL_WINDOW_FULLSCREEN;
 #else
-
-	return CIrrDeviceStub::isFullscreen();
+	return SDL_WINDOW_FULLSCREEN_DESKTOP;
 #endif
 }
 
+bool CIrrDeviceSDL::setFullscreen(bool fullscreen)
+{
+	if (!Window)
+		return false;
+	// The SDL wiki says that this may trigger SDL_RENDER_TARGETS_RESET, but
```

Comment:
can we add handlers for `SDL_RENDER_TARGETS_RESET` and `SDL_RENDER_DEVICE_RESET` and print a warning/error?
because if I understand correctly this will mess up MT